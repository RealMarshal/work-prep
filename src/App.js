import React, { Component, Fragment } from 'react';
import logo from './logo.svg';
import './App.css';
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import CounterRecompose from './CounterRecompose'
import CounterRenderProps from './CounterRenderProps'
import 'typeface-roboto'

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <Typography variant='display2' color='primary'>React App</Typography>
        </header>
        <Typography variant='headline' paragraph>
          To get started, edit <code>src/App.js</code> and save to hot-reload the page.
        </Typography>
        <Typography variant='subheading'>Counter implementation with recompose</Typography>
        <CounterRecompose />
        <Typography variant='subheading'>Counter implementation with render props</Typography>
        <CounterRenderProps
          render={(value, inc, dec) => (
            <Fragment>
              <Typography>{value}</Typography>
              <Button color='primary' onClick={inc}>Inc</Button>
              <Button color='primary' onClick={dec}>Dec</Button>
            </Fragment>
          )} 
        />
      </div>
    );
  }
}

export default App;
