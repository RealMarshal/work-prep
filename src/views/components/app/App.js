import React, { Fragment } from 'react'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'

import CounterFromHOC from '../counter/CounterFromHOC'
import Counter from '../counter/CounterRP'
import Posts from '../post/PostsWrapper'

import logo from './logo.svg'
import './App.css'
import 'typeface-roboto'

const App = () => (
  <div className="App">
    <Grid container>
      <Grid item xs={12}>
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <Typography variant='display2' color='primary'>React App</Typography>
        </header>
        <Typography variant='headline' paragraph>
          To get started, edit <code>src/App.js</code> and save to hot-reload the page.
        </Typography>
        <Typography variant='subheading'>Counter implementation with recompose</Typography>
        <CounterFromHOC />
        <Typography variant='subheading'>Counter implementation with render props</Typography>
        <Counter
          render={(value, inc, dec) => (
            <Fragment>
              <Typography>{value}</Typography>
              <Button color='primary' onClick={dec}>Dec</Button>
              <Button color='primary' onClick={inc}>Inc</Button>        
            </Fragment>
          )} 
        />
        <Posts />
      </Grid>
    </Grid>
  </div>
)

export default App;
