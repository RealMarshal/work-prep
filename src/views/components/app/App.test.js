import React from 'react';
import { shallow } from 'enzyme'
import App from './App';

describe('App', () => {

  const rendered = shallow(<App />)
  it('should render correctly', () => {
    expect(rendered).toMatchSnapshot()
  })  
})
