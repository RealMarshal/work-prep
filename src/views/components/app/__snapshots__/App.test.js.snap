// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App"
>
  <WithStyles(Grid)
    container={true}
  >
    <WithStyles(Grid)
      item={true}
      xs={12}
    >
      <header
        className="App-header"
      >
        <img
          alt="logo"
          className="App-logo"
          src="logo.svg"
        />
        <WithStyles(Typography)
          color="primary"
          variant="display2"
        >
          React App
        </WithStyles(Typography)>
      </header>
      <WithStyles(Typography)
        paragraph={true}
        variant="headline"
      >
        To get started, edit 
        <code>
          src/App.js
        </code>
         and save to hot-reload the page.
      </WithStyles(Typography)>
      <WithStyles(Typography)
        variant="subheading"
      >
        Counter implementation with recompose
      </WithStyles(Typography)>
      <withState(withHandlers(Counter)) />
      <WithStyles(Typography)
        variant="subheading"
      >
        Counter implementation with render props
      </WithStyles(Typography)>
      <Counter
        render={[Function]}
      />
      <Connect(lifecycle(PostsWrapper)) />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
